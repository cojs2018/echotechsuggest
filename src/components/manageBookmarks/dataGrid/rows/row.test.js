import React from 'react';
import { View } from 'react-native';
import render from 'react-test-renderer';

jest.mock('../cells/cell');
jest.mock('../cells/chipCell');

import Cell from '../cells/cell';
import ChipCell from '../cells/chipCell';

import Row from './row';

Cell.mockImplementation(({ value }) => <View>{value}</View>)
ChipCell.mockReturnValue(<View>''</View>)

describe('Row', () => {
    it('renders correctly', () => {
        const props = {
            rowId: 0,
            rowData: {
                bookmarkId: "2",
                articleName: "article2 - articlearchive",
                path: 'http://www.articlearchive.com/article2',
                url: new URL('http://www.articlearchive.com/article2'),
                createdAt: (new Date()).toUTCString(),
                tags: [ "tag 1" ]
            },
            onChecked: jest.fn(),
        };

        const tree = render
            .create(<Row {...props} />)
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    })
})