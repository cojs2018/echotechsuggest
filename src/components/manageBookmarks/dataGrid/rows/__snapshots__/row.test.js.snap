// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "borderStyle": "solid",
          "minHeight": 48,
          "paddingHorizontal": 16,
        },
        Object {
          "borderBottomColor": "rgba(0, 0, 0, 0.12)",
        },
        Object {},
      ],
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View>
      article2 - articlearchive
    </View>
    <View>
      Wed, 10 Mar 2021 14:05:18 GMT
    </View>
    <View>
      ''
    </View>
  </View>
</View>
`;
